configurator/src/main.rs

```diff
@@ -170,32 +170,39 @@ struct AdvancedConfig {
     bitcoin: BitcoinChannelConfig,
 }
 
-#[derive(serde::Serialize)]
-pub struct Properties<'a> {
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct Properties {
     version: u8,
-    data: Data<'a>,
+    data: Data,
 }
 
-#[derive(serde::Serialize)]
-pub struct Data<'a> {
-    #[serde(rename = "LND Sync Height")]
-    sync_height: Property<String>,
-    #[serde(rename = "Synced To Chain")]
-    synced_to_chain: Property<String>,
-    #[serde(rename = "Synced To Graph")]
-    synced_to_graph: Property<String>,
-    #[serde(rename = "LND Connect gRPC URL")]
-    lnd_connect_grpc: &'a Property<String>,
-    #[serde(rename = "LND Connect REST URL")]
-    lnd_connect_rest: &'a Property<String>,
-    #[serde(rename = "NODE URI")]
-    node_uri: &'a Property<String>,
+#[derive(serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+pub enum Data {
+    LND {
+        #[serde(rename = "LND Sync Height")]
+        sync_height: Property<String>,
+        #[serde(rename = "Synced To Chain")]
+        synced_to_chain: Property<String>,
+        #[serde(rename = "Synced To Graph")]
+        synced_to_graph: Property<String>,
+        #[serde(rename = "LND Connect gRPC URL")]
+        lnd_connect_grpc: Property<String>,
+        #[serde(rename = "LND Connect REST URL")]
+        lnd_connect_rest: Property<String>,
+        #[serde(rename = "NODE URI")]
+        node_uri: Property<String>,
+    },
+    NotReady {
+        #[serde(rename = "Not Ready")]
+        not_ready: Property<String>,
+    },
```

Comment:
I would not parameterize this arm. I would instead make this just `NotReady` and then have the serialize instance render the predefined property you are after. In fact it gives rise to the following type:
```rust
pub enum Eventual<T: Named> {
    Ready(T),
    NotReady
}
```

this allows you to write a single serialization instance for `Eventual`, that delegates the individual property render to the struct found in the type parameter.

Reply:
Moved to P2 https://github.com/Start9Labs/lnd-wrapper/issues/41